File: 206-reverse-linked-list/reverse-linked-list.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* reverseList(ListNode* head) {
        vector<int> vectorList;
        ListNode* current =  head;
        ListNode* current2 =  head;
        while(current){
            vectorList.push_back(current->val);
            current = current->next;
        }
        while (!vectorList.empty()) {
            current2->val = vectorList.back();
            vectorList.pop_back();  
            current2 = current2->next;   
        }
        return head;
    }
};

File: 206-reverse-linked-list/reverse-linked-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def reverseList(self, head: Optional[ListNode]) -> Optional[ListNode]:
        # iterative method
        cur1 = head
        reverse = ListNode()
        cur2 =  reverse
        stack = []
        while cur1:
            stack.append(cur1.val)
            cur1 = cur1.next
        print(stack)
        for i in range(len(stack)):
            cur2.next = ListNode(stack.pop())
            cur2 = cur2.next

        return reverse.next

        